src/js/stores/AppVersionsStore.js

```diff
@@ -85,7 +85,8 @@ AppDispatcher.register(function (action) {
   switch (action.actionType) {
     case AppVersionsEvents.REQUEST_VERSION_TIMESTAMPS:
       AppVersionsStore.resetOnAppChange(action.appId);
-      storeData.availableAppVersions = action.data.body.versions;
+      storeData.availableAppVersions = action.data.body.versions.sort()
```

Comment:
Please move the sorting logic to the `AppVersionsStore.getAppVersions` getter method to ensure that the app version store is always returning a well ordered list of versions. Doing so you will need to adjust the sorting _logic_ as using the current implementation would reverse the order every other time. You could use a custom sorting function as illustrated in the following, to avoid reversing the order.

```
  getAppVersions: function (appId) {
    if (appId === storeData.currentAppId) {

      return this.availableAppVersions
        .sort(function (a, b) {
          if (a < b) {
            return 1;
          }
          if (a > b) {
            return -1;
          }

          return 0;
        });
    }

    return [];
  }
```